### PR TITLE
restore customKeydownHandler on reset(RIS)

### PR DESCRIPTION
**Pull request by ayapi**
hello, thank you cool project
i noticed, my `customKeydownHandler` may be lost.
it happens after i execute `less` , `fzf`, etc.
these commands have something in common. they raise "ESC c Full Reset (RIS)".
please review this pr for fix it.

### CODE REVIEW

File: src/xterm.js [hunk 1]
```diff
@@ -3467,7 +3467,9 @@
     Terminal.prototype.reset = function() {
       this.options.rows = this.rows;
       this.options.cols = this.cols;
+      var customKeydownHandler = this.customKeydownHandler;
       Terminal.call(this, this.options);
```
Comment (Tyriar) [inline src/xterm.js]: Maybe this should be `Terminal.call(this, this.options, this.customKeydownHandler)`? Thoughts @parisk?

Comment (ayapi) [inline src/xterm.js]: thank you,
how about we add `customKeydownHandler` to `options` 's member?

Comment (Tyriar) [inline src/xterm.js]: Not sure if we would want a function to be an option?

Comment (parisk) [inline src/xterm.js]: Hmmm... not sure about which is the best way to go and I do not want to bike shed TBH. I believe that currently proposed implementation does the work and it does not break any public APIs, so I am 👍  with it.

**Comment by parisk** [comment]
Thanks a lot @ayapi for your contribution. LGTM 👍 .

**Comment by ayapi** [comment]
thank you very much, i'll use this terminal every day!

**Pull request merged**